**PA 1: Problem Scoping**

Polycystic ovary syndrome (PCOS) dataset

Link: https://www.kaggle.com/datasets/prasoonkottarathil/polycystic-ovary-syndrome-pcos

In [ ]:
import pandas as pd
from google.colab import drive

drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
# importing data
file_path_with_infertility="/content/drive/MyDrive/project assignments/datasets/PCOS_infertility.csv"
file_path_without_infertility="/content/drive/MyDrive/project assignments/datasets/PCOS_data_without_infertility.xlsx"

PCOS_inf = pd.read_csv(file_path_with_infertility)
PCOS_woinf = pd.read_excel(file_path_without_infertility, sheet_name="Full_new")

In [ ]:
PCOS_woinf.describe()

Sl. No  Patient File No.  PCOS (Y/N)   Age (yrs)  Weight (Kg)  \
count  541.000000        541.000000  541.000000  541.000000   541.000000   
mean   271.000000        271.000000    0.327172   31.430684    59.637153   
std    156.317519        156.317519    0.469615    5.411006    11.028287   
min      1.000000          1.000000    0.000000   20.000000    31.000000   
25%    136.000000        136.000000    0.000000   28.000000    52.000000   
50%    271.000000        271.000000    0.000000   31.000000    59.000000   
75%    406.000000        406.000000    1.000000   35.000000    65.000000   
max    541.000000        541.000000    1.000000   48.000000   108.000000   

       Height(Cm)          BMI  Blood Group  Pulse rate(bpm)   \
count   541.000000  541.000000   541.000000        541.000000   
mean    156.484835   24.311285    13.802218         73.247689   
std       6.033545    4.056399     1.840812          4.430285   
min     137.000000   12.417882    11.000000         13.000000   
25%     152.000000   21.641274    13.000000         72.000000   
50%     156.000000   24.238227    14.000000         72.000000   
75%     160.000000   26.634958    15.000000         74.000000   
max     180.000000   38.900000    18.000000         82.000000   

       RR (breaths/min)  ...  Pimples(Y/N)  Fast food (Y/N)  \
count        541.000000  ...    541.000000       540.000000   
mean          19.243993  ...      0.489834         0.514815   
std            1.688629  ...      0.500359         0.500244   
min           16.000000  ...      0.000000         0.000000   
25%           18.000000  ...      0.000000         0.000000   
50%           18.000000  ...      0.000000         1.000000   
75%           20.000000  ...      1.000000         1.000000   
max           28.000000  ...      1.000000         1.000000   

       Reg.Exercise(Y/N)  BP _Systolic (mmHg)  BP _Diastolic (mmHg)  \
count         541.000000           541.000000            541.000000   
mean            0.247689           114.661738             76.927911   
std             0.432070             7.384556              5.574112   
min             0.000000            12.000000              8.000000   
25%             0.000000           110.000000             70.000000   
50%             0.000000           110.000000             80.000000   
75%             0.000000           120.000000             80.000000   
max             1.000000           140.000000            100.000000   

       Follicle No. (L)  Follicle No. (R)  Avg. F size (L) (mm)  \
count        541.000000        541.000000            541.000000   
mean           6.129390          6.641405             15.018115   
std            4.229294          4.436889              3.566839   
min            0.000000          0.000000              0.000000   
25%            3.000000          3.000000             13.000000   
50%            5.000000          6.000000             15.000000   
75%            9.000000         10.000000             18.000000   
max           22.000000         20.000000             24.000000   

       Avg. F size (R) (mm)  Endometrium (mm)  
count            541.000000        541.000000  
mean              15.451701          8.475915  
std                3.318848          2.165381  
min                0.000000          0.000000  
25%               13.000000          7.000000  
50%               16.000000          8.500000  
75%               18.000000          9.800000  
max               24.000000         18.000000  

[8 rows x 42 columns]

In [ ]:
PCOS_woinf.describe()

Sl. No  Patient File No.  PCOS (Y/N)   Age (yrs)  Weight (Kg)  \
count  541.000000        541.000000  541.000000  541.000000   541.000000   
mean   271.000000        271.000000    0.327172   31.430684    59.637153   
std    156.317519        156.317519    0.469615    5.411006    11.028287   
min      1.000000          1.000000    0.000000   20.000000    31.000000   
25%    136.000000        136.000000    0.000000   28.000000    52.000000   
50%    271.000000        271.000000    0.000000   31.000000    59.000000   
75%    406.000000        406.000000    1.000000   35.000000    65.000000   
max    541.000000        541.000000    1.000000   48.000000   108.000000   

       Height(Cm)          BMI  Blood Group  Pulse rate(bpm)   \
count   541.000000  541.000000   541.000000        541.000000   
mean    156.484835   24.311285    13.802218         73.247689   
std       6.033545    4.056399     1.840812          4.430285   
min     137.000000   12.417882    11.000000         13.000000   
25%     152.000000   21.641274    13.000000         72.000000   
50%     156.000000   24.238227    14.000000         72.000000   
75%     160.000000   26.634958    15.000000         74.000000   
max     180.000000   38.900000    18.000000         82.000000   

       RR (breaths/min)  ...  Pimples(Y/N)  Fast food (Y/N)  \
count        541.000000  ...    541.000000       540.000000   
mean          19.243993  ...      0.489834         0.514815   
std            1.688629  ...      0.500359         0.500244   
min           16.000000  ...      0.000000         0.000000   
25%           18.000000  ...      0.000000         0.000000   
50%           18.000000  ...      0.000000         1.000000   
75%           20.000000  ...      1.000000         1.000000   
max           28.000000  ...      1.000000         1.000000   

       Reg.Exercise(Y/N)  BP _Systolic (mmHg)  BP _Diastolic (mmHg)  \
count         541.000000           541.000000            541.000000   
mean            0.247689           114.661738             76.927911   
std             0.432070             7.384556              5.574112   
min             0.000000            12.000000              8.000000   
25%             0.000000           110.000000             70.000000   
50%             0.000000           110.000000             80.000000   
75%             0.000000           120.000000             80.000000   
max             1.000000           140.000000            100.000000   

       Follicle No. (L)  Follicle No. (R)  Avg. F size (L) (mm)  \
count        541.000000        541.000000            541.000000   
mean           6.129390          6.641405             15.018115   
std            4.229294          4.436889              3.566839   
min            0.000000          0.000000              0.000000   
25%            3.000000          3.000000             13.000000   
50%            5.000000          6.000000             15.000000   
75%            9.000000         10.000000             18.000000   
max           22.000000         20.000000             24.000000   

       Avg. F size (R) (mm)  Endometrium (mm)  
count            541.000000        541.000000  
mean              15.451701          8.475915  
std                3.318848          2.165381  
min                0.000000          0.000000  
25%               13.000000          7.000000  
50%               16.000000          8.500000  
75%               18.000000          9.800000  
max               24.000000         18.000000  

[8 rows x 42 columns]

**PART 2.1:**

Dataset information:
> 44 features, 1 label(PCOS)
> 541 samples
> Problem: Supervised Learning

Polycystic Ovary Syndrome (PCOS) is a hormonal disorder that affects women of reproductive age. There are various factors that contribute to the development of PCOS, including insulin resistance, obesity, and elevated levels of androgen hormones. PCOS is one of the most common causes of female infertility, affecting 6% to 12% (as many as 5 million) of US women of reproductive age.

In the context of a Polycystic Ovary Syndrome (PCOS) dataset, a classification problem involves predicting whether a woman has PCOS or not, based on a set of input variables or features.

A regression problem in the context of a PCOS dataset might involve predicting a certain outcome variable based on one or more input variables. A possible regression problem on a PCOS dataset could be predicting the level of certain hormones, such as testosterone or luteinizing hormone, based on various input variables such as BMI, fasting glucose levels, and insulin resistance. The level of these hormones are crucial in identification of a different disease.

Overall, regression analysis on a PCOS dataset can be a valuable tool for understanding the relationship between different factors and the development and progression of PCOS, as well as for predicting certain outcomes based on a woman's individual characteristics.

**PART 2.2:**

A PCOS disease detection machine learning model could be very beneficial both in terms of improving healthcare outcomes and in creating business value. A machine learning model that can accurately detect PCOS could help clinicians to identify the condition earlier, allowing for earlier treatment and management of symptoms. This could improve patient outcomes, reduce the risk of long-term health complications, and potentially reduce healthcare costs.

From a business perspective, a PCOS disease detection machine learning model could have several potential applications. For example, it could be integrated into healthcare apps or digital health platforms to offer users a convenient and accessible way to detect PCOS. It could also be used by pharmaceutical companies to develop targeted treatments for PCOS or by insurance companies to identify high-risk patients and offer preventative care.

According to CB Insights' Industry Analyst Consensus, the pharmaceutical industry has experienced rapid growth in recent years and is now valued at over $1 trillion. It is anticipated that AI will be key to improving time-consuming and expensive industry tasks like disease detection and drug discovery. Therefore, PCOS disease detection AI model would be targeting a big and impactful industry.

In summary, a PCOS disease detection machine learning model could have significant benefits for patients, clinicians, and businesses, and could potentially contribute to improving healthcare outcomes and reducing healthcare costs.

This project seems to be a feasible use case since we are working with a reliable dataset with enough samples. In addition, the dataset is tabular and the training time is not as complicated as some other ML models such as computer vision and natural language processing. Despite this, the financial and non-financial benefits of this product are high due to its large impact. However, the accurate estimation of time and cost varies depending on computational resources.

**PART 2.3:**

Our problem is related to "prevent" value area since we are going to predict a disease at the early stages of disease development and prevent it from happening or worsening.

**PART 2.4:**

After a PCOS disease detection machine learning model produces results, several potential actions may be taken, depending on the specific context and intended use of the model. Here are some examples of actions that may be taken:

Confirmatory testing: If the machine learning model identifies a patient as having PCOS, further confirmatory testing may be conducted to ensure that the diagnosis is accurate. This may involve additional blood tests, ultrasounds, or other diagnostic procedures.

Early intervention and treatment: If PCOS is detected, early intervention and treatment can be initiated. This may involve lifestyle modifications, such as changes in diet and exercise, and medication to manage symptoms such as irregular menstrual cycles or insulin resistance.

Referral to a specialist: If a patient is identified as having PCOS, they may be referred to a specialist, such as an endocrinologist or gynecologist, for further evaluation and management.

Monitoring and follow-up: Patients with PCOS require ongoing monitoring to assess their response to treatment and identify any potential complications. The machine learning model may be used to monitor the patient's progress over time and adjust treatment as needed.

Research and development: If the machine learning model is used in a research context, the results may be used to inform the development of new treatments or interventions for PCOS. For example, the model may be used to identify subgroups of patients who respond particularly well to a particular treatment, or to identify new drug targets.


**PART 2.5:**

False negative result is more serious because the patient actually has the disease but the model is predicting it as a healthy. 

For correction we can ask patient to do the monitoring couple of more times and do the camparison.


**PART 2.6:**

For the classification problem: Our model will be predicting if the patient has the disear or not based on the input features.

For the regression problem: Our model will be predicting the levels of hormones based on some of the input features.

**PART 2.7:**

1. The regulatory requirements of the accuracy of a newly developed diagnosis tool to be implented into the local healthcare system. Specifically speaking, the healthcare rules and guidlines of India should be analyzed before implementing this ML tool in the market because the data is primarily collected from Indian hospitals.
2. How can we make this tool accessible to the general public that need it? Investigating the need for developing a mobile or web-based application using this ML model is crucial.


**PART 2.8:**

Ethical description:
Biases:
1. Since the data is collected from India, there might be some geographically discriminating bias if the model needs to be deployed in a different region of the world.
2. Other biases such as Age, Weight, family history, genetics, may make the training dataset imbalanced and eventually affect the model generalization performance. These biases need to be analyzed beforehand.

Red flags:
1. Data privacy is very important in this problem. Patient's data should not be made available to the users without patient's consent.
